src/ctap/mod.rs

```diff
@@ -1033,26 +1032,29 @@ where
                 versions.insert(0, String::from(U2F_VERSION_STRING))
             }
         }
-        let mut options_map = BTreeMap::new();
-        options_map.insert(String::from("rk"), true);
-        options_map.insert(
-            String::from("clientPin"),
-            self.persistent_store.pin_hash()?.is_some(),
-        );
-        options_map.insert(String::from("up"), true);
-        options_map.insert(String::from("pinUvAuthToken"), true);
-        options_map.insert(String::from("largeBlobs"), true);
+        let mut options = vec![];
```

Comment:
It's more common to write `Vec::new()` in those cases, but why not already put all the elements between line 1042 and 1055?

Reply:
We are using both `Vec::new()` and `vec![]` in our code often enough that I wouldn't want to homogenize in this PR. The reason for this initialization is that `"ep"` is the smallest key and only conditionally included. I could alternatively use `insert(0, ("ep", ...))` later.

Reply:
Sounds good for `Vec::new()` vs `vec![]`, it's just surprising.

Regarding order I first thought it was not important but apparently sorted input is important? Is there a reason other than latency? Because I don't think we should worry for sorting a few elements. Also it seems dangerous because this is a runtime requirement that can't be easily checked statically (typically with this kind of `if`-conditions).

Reply:
It is not important, but I sorted them during implementation and decided it's nice to keep. In general, inputs to `cbor_map!` etc do not have to be sorted.

Reply:
I see, sounds good.